app/views.py

```diff
@@ -54,7 +55,53 @@ def pagination(products, request):
 
 def ShopView(request):
     """View function for shop page of site."""
-    return render(request, "app/shop.html")
+    query = request.GET.get("query", "")
+    sort_option = request.GET.get("sort", "name")
+    selected_species = request.GET.getlist("species")
+    selected_categories = request.GET.getlist("categories")
+    price_max = request.GET.get('price_max', '1000')
+
+    try:
+        price_max = int(price_max)
+    except ValueError:
+        price_max = 10000
+    if price_max < 0:
+        price_max = 0
+    elif price_max > 10000:
+        price_max = 2000
```

Comment:
```suggestion
        price_max = 10000
```
như này mới hợp lý chứ nhỉ?
cơ mà có trường hợp 1 sp nào đó có giá hơn 10000 thì khi search sẽ ko tìm thấy à em?
Nên để là filter theo khoảng giá (min - max) thì sẽ hợp lý hơn, trường hợp chỉ nhập min thì sẽ filter all sp có giá >=min